jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt

```diff
@@ -0,0 +1,265 @@
+                                .sortedWith(String.CASE_INSENSITIVE_ORDER)
+                                .toList()
+                )
+
+                view.stacks.populateValues {
+                    stacks
+                }
+            }
+        }
+
+        view.stacks.addActionListener {
+            val showNewStackControls = view.stacks.selected() == getStackSelectionTextForCreateStack()
+            view.newStackNameLabel.isVisible = showNewStackControls
+            view.newStackName.isVisible = showNewStackControls
+        }
+
+        view.btnCreateS3Bucket.addActionListener {
+            // Ensure bucket creation takes place on the currently selected region
+            val currentRegionS3Client: S3Client = project.awsClient(view.region.selectedRegion!!)
+
+            val bucketDialog = CreateS3BucketDialog(
+                    project = project,
+                    s3Client = currentRegionS3Client,
+                    parent = view.content
+            )
+
+            if (bucketDialog.showAndGet()) {
+                bucketDialog.bucketName().let { newBucket -> view.s3Bucket.addAndSelectValue { newBucket } }
+            }
+        }
+    }
+
+    override fun createCenterPanel(): JComponent? = view.content
+
+    override fun getPreferredFocusedComponent(): JComponent? = view.region
+
+    override fun doValidate(): ValidationInfo? {
+        val validateSettings = validator.validateSettings(view)
+        okAction.isEnabled = validateSettings == null // We can call update settings, but not deploy lambda
+
+        return validateSettings
+    }
+
+    override fun getOKAction(): Action = deployAction
+
+    override fun createActions(): Array<Action> {
+        val actions = mutableListOf<Action>()
+        actions.add(okAction)
+        actions.add(cancelAction)
+
+        return actions.toTypedArray()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun deployServerlessApplication() {
+        if (okAction.isEnabled) {
+
+            // TODO : Iterate through the template, publishing functions to AWS (https://github.com/aws/aws-toolkit-jetbrains/issues/395)
+            Messages.showWarningDialog(
+                    project,
+                    "SAM Deployment is coming soon",
+                    "Not Implemented"
+            )
+
+            close(OK_EXIT_CODE)
+        }
+    }
+
+    private inner class DeployServerlessApplicationAction : OkAction() {
+        init {
+            putValue(Action.NAME, message("serverless.application.deploy.action.name"))
+            putValue(Action.SHORT_DESCRIPTION, message("serverless.application.deploy.action.description"))
+        }
+
+        override fun doAction(e: ActionEvent?) {
+            super.doAction(e)
+
+            if (doValidateAll().isNotEmpty()) {
+                return
+            }
+
+            deployServerlessApplication()
+
+            close(CANCEL_EXIT_CODE)
```

Comment:
Why cancel?

Reply:
Oops - this was left over from initial scaffolding